cmd/worker/Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM golang:1.20
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Copy the entire worker application
+COPY ../../ .
```

Comment:
Due to the project layout each Dockerfile requires the parent directories to build -- so the 'docker build' step must be invoked with the project root "context". I've just written everything assuming we'll be running all our scripts from the parent directory. 

E.g. if you wanted to build and run the worker service by itself (without docker-compose), you would run something like this from the root of the repo:

```bash
docker build -t worker-image -f cmd/worker/Dockerfile .
docker run -p 8081:8081 worker-image
```

Reply:
This is ok for now. I think the way to go in an industry environment (given enough time) would be to create custom scripts around docker build, so we could -> create temp folder -> copy relevant information there -> docker build inside said context -> delete temp folder

Biggest benefit is we wouldn't have all the extra files in our image and should build faster

Reply:
Made some changes: cmd/coordinator/Dockerfile
The resulting images have only the executable and a config file. Everything still works with docker compose.

I chose alpine 3.19 as the base image -- alpine was suggested by ChatGPT as a small, standard, secure linux base image.

Reply:
That's correct, alpine is usually the way to go (or -slim) since they are more lightweight which means they build faster and use less disk space